### PR TITLE
feat: Tabs close functionality and scroll

### LINKED ISSUE #32407: [Task] - IDE tabs revamp - close button and overflow scroll
Notion Doc - https://www.notion.so/appsmith/Tab-Management-499e69f9d5b74291ba40e5b309bff332
Figma - https://www.figma.com/file/DDP4RXK4YOTEJnuC8ocm47/IDE-Exploration?type=design&node-id=12264-306687&mode=design&t=KHJGKJBAKyzf6FMi-0

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR have following changes,
- Added close button and close functionality for tabs
- Removed 5 tabs limit and added scroll for tabs

Fixes #32407

## Automation

/ok-to-test tags="@tag.Sanity, @tag.IDE"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8683747982>
> Commit: d87357251bcb3a6f23942d17e548867882194295
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8683747982&attempt=1" target="_blank">Click here!</a>

## Summary by CodeRabbit

- **New Features**
	- Enhanced tab management in the IDE with new functionalities to close and remove JavaScript and query action tabs.
	- Improved user interaction within the editor interface by adding scroll behavior and a close button for each tab.

- **Bug Fixes**
	- Fixed issues related to tab rendering and close button behavior in the IDE, ensuring smoother user experience.

- **Tests**
	- Added comprehensive tests for the new features and fixes in the IDE's tab management system.

### CODE REVIEW

File: app/client/src/reducers/uiReducers/ideReducer.ts [hunk 1]
```diff
@@ -80,6 +80,30 @@ const ideReducer = createReducer(initialState, {
       showCreateModal: action.payload,
     };
   },
+  [ReduxActionTypes.CLOSE_JS_ACTION_TAB_SUCCESS]: (
```
Comment (hetunandu) [inline app/client/src/reducers/uiReducers/ideReducer.ts:83]: Why do we need to wait for success? Cant we do it immediately?

File: app/client/src/ce/sagas/JSActionSagas.ts [hunk 2]
```diff
@@ -478,3 +479,11 @@ export function* fetchJSCollectionsForViewModeSaga(
     });
   }
 }
+
+export function* closeJSActionTabSaga(
+  actionPayload: ReduxAction<{ id: string }>,
+) {
+  const id = actionPayload.payload.id;
+  yield put(closeJSActionTabSuccess({ id }));
```
Comment (hetunandu) [inline app/client/src/ce/sagas/JSActionSagas.ts:487]: Success is happening even before the redirect? Then what does this success indicate?

File: app/client/src/reducers/uiReducers/ideReducer.ts (hunk 1 above)
Comment (albinAppsmith) [reply app/client/src/reducers/uiReducers/ideReducer.ts:83]: I see the naming is confusing here... I will update this to REMOVE_QUERY_ACTION_TAB

File: app/client/src/pages/Editor/IDE/EditorTabs/FileTabs.test.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      />,
+      {
+        featureFlags: featureFlags,
+      },
+    );
+    const tabElement = getByTestId(`t--ide-tab-${mockTabs[0].title}`);
+    fireEvent.click(tabElement);
+
+    expect(mockNavigateToTab).toHaveBeenCalledWith(mockTabs[0]);
+  });
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/pages/Editor/IDE/EditorTabs/FileTabs.test.tsx:84]: This was a intentional change. Tab click should work irrelevant of feature flag.

**Review by hetunandu — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**